### PR TITLE
cubepp: Make sure width & height are > 0

### LINKED ISSUE #715: Wrong error message with invalid width 0
On linux running `vkcubepp --width 0` gives the error:
`Could not find both graphics and present queues`

Which is not true. `vkcube --width 0` prints help which seems correct behaviour.

### PR DESCRIPTION
**Pull request by charles-lunarg**
This check was present in cube but wasn't added to cubepp.

Fixes #715

### CODE REVIEW

**Review by ziga-lunarg — APPROVED** [review]
LGTM

**Review by lunarpapillo — CHANGES_REQUESTED** [review]
These changes do nothing to prevent `height` and `width` from being set to 0.  They do mean that if you explicitly pass `--width 0` or `--height 0` on the command line, the argument won't be recognized as part of the `--width` or `--height` switch, and will continue to be parsed (probably causing cryptic argument errors).

**Comment by charles-lunarg** [comment]
> These changes do nothing to prevent `height` and `width` from being set to 0. They do mean that if you explicitly pass `--width 0` or `--height 0` on the command line, the argument won't be recognized as part of the `--width` or `--height` switch, and will continue to be parsed (probably causing cryptic argument errors).

This change brings the width/height parameter into alignment with the behavior of vkcube & the gpu_number argument for both vkcube and vkcubepp. I would rather 'do what we already do consistently' than anything else.

**Comment by charles-lunarg** [comment]
> These changes do nothing to prevent height and width from being set to 0. 
There is no way to prevent that because this is a command line argument being passed in.
Passing in 0 to vkcube causes help to be printed. passing in 0 to vkcubepp will cause a crash cause the window size must be 1 or greater. Better to return help rather than crash.

**Comment by richard-lunarg** [comment]
The .cpp version declares the width and height as unsigned integers, while the C version declares them as signed integers. Thus the CPP version's scanf will never return a negative value ;-)

File: cube/cube.c [hunk 1]
```diff
@@ -4075,13 +4075,21 @@ static void demo_init(struct demo *demo, int argc, char **argv) {
...
         }
-        if (strcmp(argv[i], "--width") == 0 && i < argc - 1 && sscanf(argv[i + 1], "%d", &demo->width) == 1 && demo->width > 0) {
-            i++;
-            continue;
+        if (strcmp(argv[i], "--width") == 0 && i < argc - 1) {
+            if (sscanf(argv[i + 1], "%d", &demo->width) == 1 && demo->width > 0) {
+                i++;
+                continue;
+            } else {
+                ERR_EXIT("The --width parameter must be greater than 0", "User Error");
```
Comment (lunarpapillo) [inline cube/cube.c:4083]: If the user passes something like `--width max`, the error message will be somewhat misleading.  (I'm not sure this corner case is worth clarifying, however, though I do note that it's not much more code to make the error cases clear...)

Comment (lunarpapillo) [reply cube/cube.c:4083]: Also, there's a weird corner case if the user provides `--width` as the last parameter - no error is provoked and the width is silently unchanged.  That may be more of a concern.

**Review by lunarpapillo — COMMENTED** [review]
A couple of observations are listed.  But you should definitely consider a review from a code owner more important than anything I'd say.

File: cube/cube.c (hunk 1 above)
Comment (charles-lunarg) [reply cube/cube.c]: I've modified the code to respect whether the user has put in a number or not. If they did and its negative, it gives the correct error. If they don't provide a number, it now prints that message.

**Comment by charles-lunarg** [comment]
Bob gave in person approval (plus the issues he raised were resolved). Merging.

**Pull request merged**